service/matching/task_queue_manager.go

```diff
@@ -417,6 +419,12 @@ func (c *taskQueueManagerImpl) AddTask(
 		c.liveness.markAlive()
 	}
 
+	if c.taskQueueID.VersionSet() == dlqVersionSet {
+		// If this is the versioned task dlq, skip straight to spooling the task since we can't
```

Comment:
I don't think we should proliferate this "spooling" jargon. It isn't clear what it means. I looked at all the exported methods with Spool in the name and they just say it matches the task with a poller, but here we way that we do this because there are no pollers. I think it means we persist the task temporarily until a poller is ready (async match)? Also, I think we should say what we're skipping over here.

Reply:
:man_shrugging:  It's a pretty well-established and accurate term for what's happening. https://en.wikipedia.org/wiki/Spooling.  I think we should use it more consistently, actually